### PR TITLE
doc: add reth logo to docs

### LINKED ISSUE #3301: Add reth logo to docs
### Describe the feature

Add reth logo as:

```rust
#![cfg_attr(docsrs, feature(doc_cfg))]
#![doc(
    html_logo_url = "https://raw.githubusercontent.com/",
    html_favicon_url = "https://raw.githubusercontent.com/...",
    issue_tracker_base_url = "https://github.com/paradigmxzy/reth/issues/"
)]
```

to all crates

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by leovct**
Closes #3301.

I used the following URL for the logo: https://avatars0.githubusercontent.com/u/97369466.
edit: I couldn't find Reth logo URL so I went with Paradigm instead

```sh
find crates -type f -name 'lib.rs' -exec sh -c 'grep -q "#![cfg_attr(docsrs, feature(doc_cfg))]" "{}"; if [ $? -ne 0 ]; then echo "#![cfg_attr(docsrs, feature(doc_cfg))]\n#![doc(\n    html_logo_url = \"https://avatars0.githubusercontent.com/u/97369466?s=128\",\n    html_favicon_url = \"https://avatars0.githubusercontent.com/u/97369466?s=256\",\n    issue_tracker_base_url = \"https://github.com/paradigmxzy/reth/issues/\"\n)]" | cat - "{}" > /tmp/tempfile && mv /tmp/tempfile "{}"; fi' \;
```

### CODE REVIEW

**Comment by rkrasiuk** [comment]
this PR adds paradigm logo, we'd prob want reth's instead cc @gakonst

**Comment by leovct** [comment]
> this PR adds paradigm logo, we'd prob want reth's instead cc @gakonst

Yes, I couldn't find the Reth logo in the repo (maybe this? https://github.com/paradigmxyz/reth/blob/main/assets/reth.jpg) so I went with Paradigm logo instead. I can make the change if there's a URL that points to the Reth logo :)

**Comment by rkrasiuk** [comment]
@leovct roger. let's wait for @gakonst to chime in

**Comment by mattsse** [comment]
we have this 
https://github.com/paradigmxyz/reth/blob/main/assets/reth-alpha.png
but I feel like this ratio is not ideal for docs image because it will be rendered like this for example:

https://docs.rs/tracing/latest/tracing/

So I think we need another image with just the logo and a favicon of it

**Comment by onbjerg** [comment]
Marked as blocked since we need an image with a better aspect ratio

**Comment by onbjerg** [comment]
reth-docs

This is what it looks like

**Review by onbjerg — APPROVED** [review]

File: crates/blockchain-tree/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,9 @@
+#![cfg_attr(docsrs, feature(doc_cfg))]
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/paradigmxyz/reth/main/assets/reth-docs.png",
+    html_favicon_url = "https://avatars0.githubusercontent.com/u/97369466?s=256",
```
Comment (leovct) [inline crates/blockchain-tree/src/lib.rs:4]: @onbjerg you might want to also update this url :)

Comment (onbjerg) [reply crates/blockchain-tree/src/lib.rs:4]: I don't think it scales well to that size, so I opted to keep the Paradigm logo there. cc @gakonst for thoughts?

**Pull request merged**